frappe/public/js/frappe/form/multi_select_dialog.js

```diff
@@ -1,110 +1,63 @@
-// Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
-// MIT License. See license.txt
-
-frappe.ui.form.MultiSelectDialog = Class.extend({
-	init: function(opts) {
-		/* Options: doctype, target, setters, get_query, action */
-		$.extend(this, opts);
-
+frappe.ui.form.MultiSelectDialog = class MultiSelectDialog {
+	constructor(opts) {
+		/* Options: doctype, target, setters, get_query, action, add_filters_group, data_fields, primary_action_label */
+		Object.assign(this, opts);
 		var me = this;
-		if(this.doctype!="[Select]") {
-			frappe.model.with_doctype(this.doctype, function(r) {
+		if (this.doctype != "[Select]") {
+			frappe.model.with_doctype(this.doctype, function () {
 				me.make();
 			});
 		} else {
 			this.make();
 		}
-	},
-	make: function() {
-		let me = this;
+	}
 
+	make() {
+		let me = this;
 		this.page_length = 20;
 		this.start = 0;
+		let fields = this.get_primary_filters();
 
-		let fields = [
-			{
-				fieldtype: "Data",
-				label: __("Search Term"),
-				fieldname: "search_term"
-			},
-			{
-				fieldtype: "Column Break"
-			}
-		];
-		let count = 0;
-		if(!this.date_field) {
-			this.date_field = "transaction_date";
-		}
-
-		// setters can be defined as a dict or a list of fields
-		// setters define the additional filters that get applied
-		// for selection
-
-		// CASE 1: DocType name and fieldname is the same, example "customer" and "customer"
-		// setters define the filters applied in the modal
-		// if the fieldnames and doctypes are consistently named,
-		// pass a dict with the setter key and value, for example
-		// {customer: [customer_name]}
-
-		// CASE 2: if the fieldname of the target is different,
-		// then pass a list of fields with appropriate fieldname
-
-		if($.isArray(this.setters)) {
-			for (let df of this.setters) {
-				fields.push(df, {fieldtype: "Column Break"});
-			}
-		} else {
-			Object.keys(this.setters).forEach(function(setter) {
-				fields.push({
-					fieldtype: me.target.fields_dict[setter].df.fieldtype,
-					label: me.target.fields_dict[setter].df.label,
-					fieldname: setter,
-					options: me.target.fields_dict[setter].df.options,
-					default: me.setters[setter]
-				});
-				if (count++ < Object.keys(me.setters).length) {
-					fields.push({fieldtype: "Column Break"});
-				}
-			});
-		}
-
+		// Make results area
 		fields = fields.concat([
-			{
-				"fieldname":"date_range",
-				"label": __("Date Range"),
-				"fieldtype": "DateRange",
-			},
-			{ fieldtype: "Section Break" },
 			{ fieldtype: "HTML", fieldname: "results_area" },
-			{ fieldtype: "Button", fieldname: "more_btn", label: __("More"),
-				click: function(){
-					me.start += 20;
+			{
+				fieldtype: "Button", fieldname: "more_btn", label: __("More"),
+				click: () => {
+					this.start += 20;
 					frappe.flags.auto_scroll = true;
```

Comment:
If there is no need of this auto_scroll, let's remove it

Reply:
removed